### ISSUE TITLE
unused variables

### ISSUE THREAD

**Issue by eclare108213** [CICE-Consortium/Icepack#120]
Please would someone using a gnu or nag compiler produce a list of warnings with their module/line locations?   The priority here is getting unused variables out of the columnphysics interface subroutine argument lists, so I'm most interested in the "Unused" warnings from the list below.  But it would be good to capture all of them, to clean up at some point.   

**GNU Warnings**
Unused variable 
Creating array temporary 
Label
Unused dummy argument 
Unused module variable 
Unused parameter 
Unused PRIVATE module variable 
Obsolescent feature 
Conversion 
DO loop 
Equality comparison 
Inequality comparison 

**Nag Warnings**
Explicitly imported into 
Unused dummy variable 
Unused local variable 
DO loop has zero iteration 
Unused PARAMETER 
Initialised but never used 
Character data-value padded 
Array section passed to old style dummy 
Expression in IF construct is constant 
Initialisation expression for 
Expression in logical IF is constant 
Set but never referenced

**Comment by apcraig** [comment]
The gnu compiler is not generating any warning messages on gordon on my current args branch.  We should try nag.

**Comment by dabail10** [comment]
I have the code compiling with NAG now. Lots of warning messages! In pretty much every file there are "use icepack_warnings" that are not actually used. Also, the "subname" variable is set nearly everywhere and hardly used. Are you fine with me removing all of these?

**Comment by apcraig** [comment]
I think we want subname in every subroutine.  That was a huge effort to do.  It's nice that it's done and will be handy as we move forward.  I would say the same about use warnings.  I don't think it hurts that it's in every module, but am fine if we want to remove those from modules where it's not being used.  But please don't remove the subnames.

**Comment by eclare108213** [comment]
I agree.  Let's keep the subnames for future use, as needed.  Can you identify just variables that are passed through subroutine calls, for now?

**Comment by dabail10** [comment]
Here is all of warnings that are not "subname" or icepack_warning. There are a lot here. I am happy to work on this, but I will need guidance on a few.

[unused.txt](https://github.com/CICE-Consortium/Icepack/files/1696243/unused.txt)

**Comment by eclare108213** [comment]
Wow, there are lots.  I think you could just start down the list, deleting.  You might prioritize "dummy" over "local" if indeed dummy means subroutine argument.  I'm guessing that "explicitly imported" means in a use statement.  Those and unused locals can be cleaned up later.  Do loops are likely of the form do n=1,0 where the 0 is because of an inactive tracer; these need to be wrapped in the appropriate conditional but for now are probably harmless (low priority).  Thanks Dave, I'm delighted for you to work on it.
